src/apps/config/leader.lua

```diff
@@ -169,8 +169,9 @@ local function path_setter_for_grammar(grammar, path)
       assert(grammar.type == 'struct')
       local tail_id = data.normalize_id(tail_name)
       return function(config, subconfig)
-         getter(config)[tail_id] = subconfig
-         return config
+	 local newconfig = lib.deepcopy(config)
```

Comment:
So I talk about the app graph but really I'm only concerned with the configuration, it just so happens I can get the configuration from the app graph. I don't believe there is any information passed to the `compute_update_actions` function (called from `Leader:update_configuration`) that has the previous name. The current app graph and the new app graph both which are passed in actually already have the new name applied to them. My thinking here is that the `current_app_graph` should have the old name until [this happens](https://github.com/Igalia/snabb/blob/lwaftr/src/apps/config/leader.lua#L414).

It's correct that setting the name causes nothing to restart, no app besides the leader should be concerned about the name change. At the moment no app cares what the name is and thus the name can be changed without consulting the follower at all, even to share the information that it has been changed.

So before I did have it as a global variable accessible as the name property on `core.app.configuration`. I changed it in the PR as I didn't want potentially stale information kept there on other processes who didn't get the update. I feel like that data is somewhat incidental with relation to this. Whilst it does have the app name I could use, so does the leader's config if it were not mutated but rather created anew and then set once those changes have been queued for the follower.

The follower does not need the name, correct.